test/tests.zig

```diff
@@ -1037,11 +1037,13 @@ pub const StandaloneContext = struct {
         requires_macos_sdk: bool = false,
         requires_stage2: bool = false,
         use_emulation: bool = false,
+        requires_symlinks: bool = false,
     }) void {
         const b = self.b;
 
         if (features.requires_macos_sdk and !self.enable_macos_sdk) return;
         if (features.requires_stage2 and self.omit_stage2) return;
+        if (features.requires_symlinks and builtin.os.tag == .windows) return;
```

Comment:
I do not think this is correct. I run these tests locally on both my arm64 and x86 Windows installations with Developer Mode enabled thereby having symlinks capability enabled by default and thus have no problem passing MachO tests on Windows. A better solution to this would be to require the user to declare they have symlinking enabled on Windows via a build flag such as `-Denable-symlinks-windows`. Ideally however we would find a way around the symlinks altogether.

Also please note that testing the MachO linker on Windows (or any foreign host) is of paramount importance to weed out any regressions in cross-linking to MachO from those hosts, i.e., targeting macOS from should just work.

Reply:
I think detecting if symlinks are viable might be tricky = the only thing I can think of is trying to create a dummy symlink and checking for error condition. I believe having a declarative mechanism where the user declares a promise they have symlinks rights like I described above is the way to go. We do the same with Rosetta2 on arm macs, and the way this works it that in `build.zig` in the root of the project, we declare a build option that defaults to false:

```zig
// build.zig
const enable_macos_sdk = b.option(bool, "enable-macos-sdk", "Run tests requiring presence of macOS SDK and frameworks") orelse false;
```

If you trace it all the way to the test harness, this should give you a good idea how this mechanism works.